### PR TITLE
Backport fix for loadbalancer deadlock

### PR DESCRIPTION
**Pull request by urso**
Backport of #1832

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -26,13 +26,8 @@ https://github.com/elastic/beats/compare/v1.2.3...1.3[Check the HEAD diff]
 ==== Bugfixes
 
 *Affecting all Beats*
+- Fix beats load balancer deadlock if max_retries: -1 or publish_async is enabled in filebeat. {issue}1829[1829]
 - Fix output modes backoff counter reset. {issue}1803[1803] {pull}1814[1814] {pull}1818[1818]
-
-- Drain response buffers when pipelining is used by redis output. {pull}1353[1353]
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: Did you remove that part here on purpose?

**Comment by ruflin** [comment]
@urso Needs rebase

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (urso) [inline CHANGELOG.asciidoc]: Yes, cherry picking manager to add too many entries in Changelog.

**Comment by urso** [comment]
@ruflin rebased.

**Comment by ruflin** [comment]
@urso Not sure if there is an easy way to add a test for this to make sure we don't reintroduce it.

**Pull request merged**